### DISCUSSION TITLE
Weekly Pumpernickle (3/9 - 3/16)

### DISCUSSION THREAD

**Discussion by mgatelabs** [mgatelabs/GenshinSolvers#120] [General]
New discussion time

**Comment by stefnotch** [comment]
It seems like clicking on the cubes doesn't work at the moment? (In the deployed version)

**Reply by mgatelabs** [reply]
I see what is going on.  If you navigate to a puzzle from the main map/list, it works, but if you start to navigate to other puzzles via the bottom map, something gets messed up.  I think we only setup the interaction manager once for the "starting" cubes and never updated it on switch

**Reply by mgatelabs** [reply]
Got it

**Comment by mgatelabs** [comment]
Seirai Island is almost complete, I need to finish off that puzzle that requires going between islands.  Think I'll use my kids accounts to get the raw details, since it is complicated.

**Comment by mgatelabs** [comment]
Seirai Island is complete and it was weird, but x16779 it is actually two map markers, so I made up a system so they can be linked together.

**Comment by mgatelabs** [comment]
4 islands are now complete.  Now all that is left is fish and scary island.

**Reply by mgatelabs** [reply]
Also finally had to implement a puzzle with a custom solution.

**Reply by stefnotch** [reply]
Woah, that's amazing to hear!
The fish island has that one very unique puzzle. I wonder how that one works, since I've never actually solved it.

**Comment by mgatelabs** [comment]
Thinking, should a tracker be placed in the deployed page?  Like the google web tool thing, so we can see how many people are using it?

**Reply by stefnotch** [reply]
Sounds like a reasonable decision. Though, preferably a tracker that actually respects privacy and doesn't require us to deal with the GDPR or other data protection laws. I'm thinking something simple that only logs page hits might do the trick.

**Reply by mgatelabs** [reply]
I've been looking around and there isn't a great solution for basic tracking, it needs a 3rd party service and google isn't GDPR compliant, but with some tweeks you could disable most of the tracking, but still wouldn't want to give the big G any more insight.

**Reply by stefnotch** [reply]
I'll ask someone who might have a bit of experience in that area

**Reply by stefnotch** [reply]
They recommended https://plausible.io/ , which seems to be pretty decent at a first glance

**Reply by mgatelabs** [reply]
It seems good, except for the $9 a month for 10k traffic.

**Reply by stefnotch** [reply]
I see, that's definitely not ideal for a page with zero revenue.

**Comment by mgatelabs** [comment]
Just finished adding the 3 special Watatsumi Island puzzles.  Had to add in a new way to specify the "solution", since you have to hit the pieces in a certain order.

**Comment by mgatelabs** [comment]
Been a while, exploring a cave, and I'm trying to write a Old Chou's Treasure solver.

**Reply by stefnotch** [reply]
Oh, is it essentially a variant of Minesweeper?
If yes, then that sounds like a fun thing to write a solver for. (Here is one example of a best-possible solver: https://www.youtube.com/watch?v=cGUHehFGqBc )

**Comment by mgatelabs** [comment]
![Screenshot 2022-05-04 213734](https://user-images.githubusercontent.com/6304749/166856268-efd56952-7e24-4dcc-887d-2c78b7948f78.png)

This is what I have for the solver, you click and you can set the tile to be Empty, Cabbage or Iron

**Reply by stefnotch** [reply]
Oh that's lovely! First bit of feedback is that it might be nice to explain "empty means no bombs nearby, cabbage means..."

**Reply by stefnotch** [reply]
![image](https://user-images.githubusercontent.com/10220080/166876010-a8a477d8-4162-444f-a43f-f82a1ea7ba8b.png)

Also, I guess the solver isn't quite finished yet? If I didn't misunderstand the rules, right here it should be able to mark the topmost/rightmost ?BOMB? tiles as SAFE.

**Reply by mgatelabs** [reply]
Right now i'm doing simple matching, if I kept track of how many squares want that tile to be a bomb then I could eliminate the top and right ones, since the bottom and left have a stronger weight.

**Reply by mgatelabs** [reply]
Ok, re-wrote the algorithm and now it detects more edge cases.

**Comment by mgatelabs** [comment]
https://mgatelabs.github.io/GenshinSolvers/oldchoutreasure
